RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -10,22 +10,41 @@ import SwiftUI
 
 #if PAYWALL_COMPONENTS
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
-struct TemplateComponentsView: View {
-
-    @Environment(\.locale)
-    var locale
+public struct TemplateComponentsView: View {
 
     let paywallComponentsData: PaywallComponentsData
+    let componentViewModels: [PaywallComponentViewModel]
 
-    init(paywallComponentsData: PaywallComponentsData) {
+    public init(paywallComponentsData: PaywallComponentsData, offering: Offering) {
         self.paywallComponentsData = paywallComponentsData
+
+        let components = paywallComponentsData.componentsConfig.components
+        self.componentViewModels = components.map { component in
+
+            // STEP 1 - Get list of preferred locales (and default)
+
+            // STEP 2 - Get localization for one of preferred locales in order
+            let chosenLocale = Locale.current
+            // TOOD: This logic is so wrong
+            let localizations = paywallComponentsData.componentsLocalizations
+            let localization = localizations[chosenLocale.identifier] ??
+                                  localizations.values.first ??
+                                  [String: String]()
```

Comment:
I think this is probably addressed in the next PR but we should/will be erroring out and not defaulting to an empty dictionary if no locales found 😇

Reply:
In the next PR it logs an error and returns an empty dictionary. It will then error out when if it tries to localize a string on any component. I guess we can discuss in the next PR!